### ISSUE TITLE
Update broke Github updating

### ISSUE THREAD

**Issue by stephendgilbert** [cs50/codespace#78]
Unfortunately, this couldn't have happened at a worse time for me. Our classes start on Monday. We have been using CodeSpaces but I have used your CS50 devcontainer.json file so it looks like the CS50 IDE. Everything worked fine until we were prompted to update. When that happened, git stopped authenticating. This dialog pops up when it tries to automatically update the repo.
![image](https://user-images.githubusercontent.com/15840496/215221259-90cc2e38-2de7-4c5a-84dd-62eff971c5e0.png)

When you go to the Github logs, this error message appears:

> 2023-01-27 22:44:46.704 [info] > git cat-file -s 66ba7cbbbf34af3f81c6850e635b9b6846f7afae [1ms]
> 2023-01-27 22:44:46.792 [info] > git show --textconv :cpp-hello.cpp [2ms]
> 2023-01-27 22:44:47.084 [info] > git push --force origin main [1270ms]
> 2023-01-27 22:44:47.084 [info] remote: Invalid username or password.
> fatal: Authentication failed for 'https://github.com/occ-cs150/cs150-s23-sgilbert-student/'
> 2023-01-27 22:49:30.024 [info] > git log --oneline --cherry main...main@{upstream} -- [2ms]

Before the update, this is what happened in the github log.

> git cat-file -s 2db30975169a6164ca90766435576e4d7f34cb28 [1ms]
> 2023-01-27 22:35:21.416 [info] > git show --textconv :cs150/hw/h01/h01.cpp [3ms]
> 2023-01-27 22:35:22.268 [info] > git push --force origin main [1766ms]
> 2023-01-27 22:35:22.268 [info] To https://github.com/occ-cs150/cs150-s23-steveandkathyg
>    9f1606d..2501985  main -> main
> 2023-01-27 22:35:22.289 [info] > git config --get commit.template [9ms]
> 

Do you have any hints for me that would help me get going on Monday?

**Comment by rongxin-liu** [comment]
Please ensure you and your students access your codespace via https://code.cs50.io and authorize any newly requested permissions. Please also try visiting https://code.cs50.io/restart to restart your Codespaces and see if that worked.

What happened with the recent updates was code.cs50.io will now set an encrypted OAuth token in the Codespace environment automatically, and all git authentications would go through that token by default. That said, you were getting the authentication error was likely because the token was not set or the token did not have the required permission.

**Comment by stephendgilbert** [comment]
Thanks. We are not using code.cs50.io. Last summer Professor Malan encouraged us just to use the CodeSpaces (directly from Github), so that it would work with Github classroom (which code.cs50.io does not). He also said, when we set up our codespace repository for the students, to use the .devcontainer from CS50. That has worked fine until now. We could go back to using code.cs50.io, but then we have to have the students upload all of the assignments, and unlike github classroom, the instructors can't check the student's repositories.

**Comment by dmalan** [comment]
Hi @stephendgilbert, ah, I think we can fix this. Will loop back shortly!

**Comment by stephendgilbert** [comment]
Thanks so much.

**Comment by dmalan** [comment]
Hi @stephendgilbert, apologies for the last-minute stressor! My fault, afraid I didn't consider the scenario where someone's using our `.devcontainer.json` but not our https://code.cs50.io entrypoint. Once https://github.com/cs50/codespace/pull/79 is merged and deployed (e.g., by morning), you should be able to run `update50` again to fix things. In the meantime, you can manually fix your own codespace with:

```
sudo wget -O /opt/cs50/bin/gitcredential_github.sh https://raw.githubusercontent.com/cs50/codespace/f081c4d459e23a46bd9c8569e2cec4e1add53741/opt/cs50/bin/gitcredential_github.sh
```

Alternatively, funny enough, the recent change @rongxin-liu mentions actually makes it possible to use GitHub Classroom within https://code.cs50.io now. Previously, the default `$GITHUB_TOKEN` only had access to @code50 repos. The new `$CS50_TOKEN` has access to any repo that the logged-in user has access to. Organizations (e.g., the one you're using with Classroom) have to explicitly opt into such access, though, per https://cs50.readthedocs.io/code/#organization-access.

And note that, if you do want students to be able to clone a Classroom repo into https://code.cs50.io, they'll still have to do it in, e.g., `/workspaces`, since `/workspaces/$RepositoryName` is already a Git repo, per https://cs50.readthedocs.io/code/#git.

Upside of https://code.cs50.io, though, is that students then have a "monocodespace" in which they can store multiple assignments.

**Comment by stephendgilbert** [comment]
Thanks so, so much. I actually used the code.cs50.io during my class last summer, and got the idea for using the Codespaces from you at the summer. I did that and it worked fine during the Fall semester. It would not be that difficult to change back to using code.cs50.io. I just tried the fix you implemented on a couple of my codespaces (and the instructor whom I'm walking through the process) and it worked fine. Worst case, I'll just add the line to the config script we have the students run when they first set up their codespace.
Again, thanks so very much.
